From 97f9ef30945dff91cd2f95ae9758fad99bbb25ee Mon Sep 17 00:00:00 2001
From: Joonas Lahtinen <joonas.lahtinen@linux.intel.com>
Date: Tue, 9 Feb 2021 15:51:25 +0200
Subject: [PATCH 1/2] nightly.conf: Remove deprecated "drm-intel-next-queued"

Acked-by: Jani Nikula <jani.nikula@intel.com>
Signed-off-by: Joonas Lahtinen <joonas.lahtinen@linux.intel.com>
---
 nightly.conf | 1 -
 1 file changed, 1 deletion(-)

diff --git a/nightly.conf b/nightly.conf
index cca1e1d3c8e2..2d7c6c7d1eaa 100644
--- a/nightly.conf
+++ b/nightly.conf
@@ -85,7 +85,6 @@ drm_tip_config=(
 
 	"drm-misc		drm-misc-next"
 	"drm-intel		drm-intel-next"
-	"drm-intel		drm-intel-next-queued"
 	"drm-amd		drm-amd-next"
 
 	"sound-upstream		for-linus"
-- 
2.29.2

