diff --git a/drivers/gpu/drm/i915/i915_gem_evict.c b/drivers/gpu/drm/i915/i915_gem_evict.c
index 1441c719277c..60ca4f05ae94 100644
--- a/drivers/gpu/drm/i915/i915_gem_evict.c
+++ b/drivers/gpu/drm/i915/i915_gem_evict.c
@@ -213,6 +213,9 @@ i915_gem_evict_something(struct i915_address_space *vm,
 		if (I915_SELFTEST_ONLY(igt_evict_ctl.fail_if_busy))
 			return -EBUSY;
 
+		ret = ggtt_flush(dev_priv);
+		if (ret)
+			return ret;
 
 		cond_resched();
 		goto search_again;
