diff --git a/drivers/gpu/drm/i915/i915_debugfs.c b/drivers/gpu/drm/i915/i915_debugfs.c
index b418a11a20bd..252c4bcd7348 100644
--- a/drivers/gpu/drm/i915/i915_debugfs.c
+++ b/drivers/gpu/drm/i915/i915_debugfs.c
@@ -2929,10 +2929,10 @@ static int pipe_crc_set_source(struct drm_device *dev, enum pipe pipe,
 		DRM_DEBUG_DRIVER("stopping CRCs for pipe %c\n",
 				 pipe_name(pipe));
 
-		mutex_lock(&crtc->base.mutex);
+		drm_modeset_lock(&crtc->base.mutex, NULL);
 		if (crtc->active)
 			intel_wait_for_vblank(dev, pipe);
-		mutex_unlock(&crtc->base.mutex);
+		drm_modeset_unlock(&crtc->base.mutex);
 
 		spin_lock_irq(&pipe_crc->lock);
 		entries = pipe_crc->entries;
