diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index cefbe6fc62bf..ed5f140d1e17 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -4125,13 +4125,9 @@ static void haswell_crtc_enable(struct drm_crtc *crtc)
 	/* If we change the relative order between pipe/planes enabling, we need
 	 * to change the workaround. */
 	haswell_mode_set_planes_workaround(intel_crtc);
-<<<<<<< HEAD
 	intel_crtc_enable_planes(crtc);
-=======
-	ilk_crtc_enable_planes(crtc);
 
 	drm_vblank_on(dev, pipe);
->>>>>>> origin/topic/drm-vblank-rework
 }
 
 static void ironlake_pfit_disable(struct intel_crtc *crtc)
@@ -4638,8 +4634,9 @@ static void valleyview_crtc_enable(struct drm_crtc *crtc)
 	for_each_encoder_on_crtc(dev, crtc, encoder)
 		encoder->enable(encoder);
 
-<<<<<<< HEAD
 	intel_crtc_enable_planes(crtc);
+
+	drm_vblank_on(dev, pipe);
 }
 
 static void i9xx_set_pll_dividers(struct intel_crtc *crtc)
@@ -4649,9 +4646,6 @@ static void i9xx_set_pll_dividers(struct intel_crtc *crtc)
 
 	I915_WRITE(FP0(crtc->pipe), crtc->config.dpll_hw_state.fp0);
 	I915_WRITE(FP1(crtc->pipe), crtc->config.dpll_hw_state.fp1);
-=======
-	drm_vblank_on(dev, pipe);
->>>>>>> origin/topic/drm-vblank-rework
 }
 
 static void i9xx_crtc_enable(struct drm_crtc *crtc)
@@ -4719,11 +4713,9 @@ static void i9xx_crtc_enable(struct drm_crtc *crtc)
 	for_each_encoder_on_crtc(dev, crtc, encoder)
 		encoder->enable(encoder);
 
-<<<<<<< HEAD
 	intel_crtc_enable_planes(crtc);
-=======
+
 	drm_vblank_on(dev, pipe);
->>>>>>> origin/topic/drm-vblank-rework
 }
 
 static void i9xx_pfit_disable(struct intel_crtc *crtc)
@@ -7538,36 +7530,6 @@ static bool haswell_get_pipe_config(struct intel_crtc *crtc,
 	return true;
 }
 
-<<<<<<< HEAD
-=======
-static int intel_crtc_mode_set(struct drm_crtc *crtc,
-			       int x, int y,
-			       struct drm_framebuffer *fb)
-{
-	struct drm_device *dev = crtc->dev;
-	struct drm_i915_private *dev_priv = dev->dev_private;
-	struct intel_encoder *encoder;
-	struct intel_crtc *intel_crtc = to_intel_crtc(crtc);
-	struct drm_display_mode *mode = &intel_crtc->config.requested_mode;
-	int ret;
-
-	ret = dev_priv->display.crtc_mode_set(crtc, x, y, fb);
-
-	if (ret != 0)
-		return ret;
-
-	for_each_encoder_on_crtc(dev, crtc, encoder) {
-		DRM_DEBUG_KMS("[ENCODER:%d:%s] set [MODE:%d:%s]\n",
-			encoder->base.base.id,
-			drm_get_encoder_name(&encoder->base),
-			mode->base.id, mode->name);
-		encoder->mode_set(encoder);
-	}
-
-	return 0;
-}
-
->>>>>>> origin/topic/drm-vblank-rework
 static struct {
 	int clock;
 	u32 config;
