diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index 22f75a3..6db833e 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -3956,10 +3956,7 @@ int skl_check_pipe_max_pixel_rate(struct intel_crtc *intel_crtc,
 		DRM_DEBUG_KMS("Max supported pixel clock with scaling exceeded\n");
 		return -EINVAL;
 	}
-<<<<<<< HEAD
-=======
 
->>>>>>> drm-upstream/drm-next
 	return 0;
 }
 
