diff --git a/sound/isa/sb/emu8000.c b/sound/isa/sb/emu8000.c
index 3b379dc..d56973b 100644
--- a/sound/isa/sb/emu8000.c
+++ b/sound/isa/sb/emu8000.c
@@ -1138,11 +1138,7 @@ snd_emu8000_new(struct snd_card *card, int index, long port, int seq_ports,
 		snd_emu8000_free(hw);
 		return err;
 	}
-<<<<<<< HEAD
-#if IS_REACHABLE(CONFIG_SND_SEQUENCER)
-=======
 #if IS_ENABLED(CONFIG_SND_SEQUENCER)
->>>>>>> sound-upstream/for-linus
 	if (snd_seq_device_new(card, index, SNDRV_SEQ_DEV_ID_EMU8000,
 			       sizeof(struct snd_emu8000*), &awe) >= 0) {
 		strcpy(awe->name, "EMU-8000");
