diff --git a/drivers/gpu/drm/i915/gt/intel_breadcrumbs.c b/drivers/gpu/drm/i915/gt/intel_breadcrumbs.c
index 429a463b637c..00918300f53f 100644
--- a/drivers/gpu/drm/i915/gt/intel_breadcrumbs.c
+++ b/drivers/gpu/drm/i915/gt/intel_breadcrumbs.c
@@ -199,13 +199,6 @@ slist_add(struct llist_node *node, struct llist_node *head)
 	return node;
 }
 
-static struct llist_node *
-slist_add(struct llist_node *node, struct llist_node *head)
-{
-	node->next = head;
-	return node;
-}
-
 static void signal_irq_work(struct irq_work *work)
 {
 	struct intel_breadcrumbs *b = container_of(work, typeof(*b), irq_work);
